pydash/pydash_app/dashboard/aggregator/aggregator_group.py

```diff
@@ -177,16 +212,187 @@ def fetch_aggregator(self, partition_field_names):
             - Group-by:
               * 'group_by' - e.g. None
 
-          Note that when providing two filters of the same type, a
+          Note that when providing two filters of the same type, a ValueError is raised.
 
         :return: An Aggregator instance that contains the right aggregated data for this query.
           Note that if an invalid value is given, a new (and empty) Aggregator is returned, due to the lazy addition.
         """
         try:
-            partition = self.partition_names[frozenset(partition_field_names.keys())]
+            partition = self.partition_names[frozenset(filter_dict.keys())]
         except KeyError:
             # TODO: discern what exactly goes wrong higher up in the callback chain.
             raise ValueError("Bad input value: input filter type is not supported,"
                              " input filter-value not formatted correctly,"
                              " or multiple input filters of the same type.")
-        return self.partitions[partition][frozenset(partition_field_names.values())]
+        return self.partitions[partition][frozenset(filter_dict.values())]
+
+    def fetch_aggregator_daterange(self, filters, datetime_begin, datetime_end):
+        """
+        Fetches an aggregator over the entire provided datetime range. Note that filters may not contain time-based
+        properties.
+        :param filters: A dictionary that contains property_name-value pairs to filter on.
+          This is in the gist of {'ip': '127.0.0.1', 'version': '1.0.1'}
+          For the complete set of possible filters, see AggregatorGroup.fetch_aggregator.
+        :param datetime_begin: A datetime object indicating the inclusive lower bound for the datetime range to
+         aggregate over.
+        :param datetime_end:  A datetime object indicating the exclusive upper bound for the datetime range to
+         aggregate over.
+        :return: An Aggregator object that contains the aggregated data over the entirety of the specified datetime
+         range.
+        """
+
+        for key in filters.keys():
+            if key in ['year', 'month', 'week', 'day', 'hour', 'minute']:
+                raise ValueError('filters may not contain time-based properties')
+
+        date_chunks = chop_date_range_into_chunks(datetime_begin, datetime_end)
+        aggregator = Aggregator()
+
+        for key, value in date_chunks.items():
+            filters_cpy = copy(filters)
+            for datetime in value:
+                filters_cpy[key] = datetime.strftime(datetime_formats[key])
+                aggregator += self.fetch_aggregator(filters_cpy)
+
+        return aggregator
+
+
+def chop_date_range_into_chunks(datetime_begin, datetime_end):
```

Comment:
I'm not a big fan of the formatting of this function; a lot of whitespace that makes it a bit hard to read everything. Maybe decrease the indentation level a bit?

Reply:
This is just aesthetic though, the functionality looks good :+1:

Reply:
The reason why I have so much whitespace is that I tried to keep everything structured, while trying not to write very long lines of code.

Reply:
In this case I'd say writing a couple lines more would be preferable to having lines that are 99% whitespace :wink: